beacon.go

```diff
@@ -17,110 +17,185 @@
+	table     string
 }
 
 var config = beaconConfig{
-	variantSetIds: []string{"3049512673186936334"},
+	projectID: os.Getenv("GOOGLE_CLOUD_PROJECT"),
+	table:     os.Getenv("TABLE"),
 }
 
 func init() {
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
+	if err := validateServerConfig(); err != nil {
+		writeError(w, err)
+		return
 	}
 
-	coord, err := strconv.ParseInt(r.FormValue("coordinate"), 10, 64)
+	refName, allele, coord, err := parseInput(r)
 	if err != nil {
-		return nil, "", err
+		writeError(w, err)
+		return
 	}
 
-	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
-	search.Start = coord
-	search.End = search.Start + 1
-
-	if search.ReferenceName == "" {
-		return nil, "", errors.New("missing reference name")
-	}
-	if allele == "" {
-		return nil, "", errors.New("missing allele")
+	ctx := appengine.NewContext(r)
+	exists, err := genomeExists(ctx, refName, allele, coord)
+	if err != nil {
+		writeError(w, err)
+		return
 	}
 
-	return search, allele, nil
+	if err := writeResponse(w, exists); err != nil {
+		writeError(w, err)
+		return
+	}
 }
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	c := appengine.NewContext(r)
-	search, allele, err := requestToSearch(r)
+func genomeExists(ctx context.Context, refName string, allele string, coord int64) (bool, error) {
+	bqclient, err := bigquery.NewClient(ctx, config.projectID)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		return false, newDataAccessError("Creating a BigQuery client", err)
 	}
 
-	client, err := google.DefaultClient(c, genomics.GenomicsReadonlyScope)
+	// Start is inclusive, End is exclusive.  Search exactly for coordinate.
+	query := fmt.Sprintf(`
+		SELECT count(v.reference_name) as count
+		FROM %s as v
+		WHERE reference_name='%s'
+			AND v.start <= %d AND %d < v.end
+	 	 	AND reference_bases='%s'
+		LIMIT 1`,
+		fmt.Sprintf("`%s`", config.table),
+		refName,
+		coord,
+		coord+1,
+		allele)
+	q := bqclient.Query(query)
+
+	it, err := q.Read(ctx)
 	if err != nil {
-		http.Error(w, "Invalid server configuration", http.StatusInternalServerError)
+		return false, newDataAccessError("Querying database", err)
+	}
+
+	type Result struct {
+		Count int
+	}
+	var result Result
+	if err := it.Next(&result); err != nil {
+		return false, newDataAccessError("Reading query result", err)
+	}
+	return result.Count > 0, nil
+}
+
+func validateServerConfig() error {
+	if config.projectID == "" {
+		return newInvalidConfigError("validating GOOGLE_CLOUD_PROJECT value", errors.New("value is mandatory"))
+	}
+	if config.table == "" {
+		return newInvalidConfigError("validating TABLE value", errors.New("value is mandatory"))
```

Comment:
Something simpler like

fmt.Errorf("%s must be specified", tableConfigurationKey)

will be sufficient since it will result in a generic server error.  We only need to be specific about error codes where it makes sense and is defined by the specification.

(Applies to the project too)

Reply:
done